From 05dfd1426c16b30595df7d978b9fdc9c952a18e7 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 10 Apr 2021 13:07:15 +0200
Subject: [PATCH 1721/1936] UI: Remove duplicate OscilGen prepare call

The phase/mag ports already call prepare, no reason to call it twice.
---
 src/UI/OscilGenUI.fl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/UI/OscilGenUI.fl b/src/UI/OscilGenUI.fl
index d67bed0e..e164536b 100644
--- a/src/UI/OscilGenUI.fl
+++ b/src/UI/OscilGenUI.fl
@@ -149,7 +149,6 @@ class Oscilharmonic {: {public Fl_Group}
     phase->value(64);
     phase->selection_color(0);
  }
- o->osc->requestValue(o->loc+"prepare");
  o->osc->requestValue(o->loc+"spectrum");
  o->osc->requestValue(o->loc+"waveform");
 
@@ -162,7 +161,6 @@ display->redraw();}
       Fl_Slider phase {
         callback {
 o->osc->writeValue(o->loc+"phase"+to_s(n), (char) o->value());
-o->osc->requestValue(o->loc+"prepare");
 o->osc->requestValue(o->loc+"spectrum");
 o->osc->requestValue(o->loc+"waveform");
 
@@ -259,6 +257,8 @@ class OscilEditor {open : {public PresetsUI_}
         Fl_Button applybutton {
           label Apply
           callback {
+          //call PADnoteParameters' "prepare" port, which computes the whole
+          //PAD synth wave table
           dummy->osc->requestValue(loc+"../prepare");}
           xywh {305 285 60 20} box THIN_UP_BOX labelfont 1
           code0 {if(adnotep) o->hide();}
-- 
2.47.0

